Sources/AwaitKit.swift

```diff
@@ -73,5 +95,62 @@ public func await<T>(_ body: @escaping () throws -> T) throws -> T {
  */
 @discardableResult
 public func await<T>(_ promise: Promise<T>) throws -> T {
-  return try AwaitKit.awaitQueue.await(promise)
+    return try await(promise, queue: awaitQueue)
+}
+
+/**
+ Awaits that the given closure finished on the receiver and returns its value or throws an error if the closure failed.
+ 
+ - parameter body: The closure that is executed on the receiver.
+ - throws: The error sent by the closure.
+ - returns: The value of the closure when it is done.
+ - seeAlso: await(promise:)
+ */
+@discardableResult
+public func await<T>(_ body: @escaping () throws -> T, queue: DispatchQueue) throws -> T {
```

Comment:
I've read that proposal but when I've did that rename Swift complained that the use of those calls are ambiguous, but feel free to change it to what other option fits you.